xtable-core/src/main/java/org/apache/xtable/parquet/ParquetStatsExtractor.java

```diff
@@ -0,0 +1,115 @@
+
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.Encoding;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.hadoop.util.HiddenFileFilter;
+import org.apache.parquet.io.ParquetDecodingException;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.Encoding;
+import org.apache.parquet.column.statistics.Statistics;
+import java.util.stream.Collectors;
+import org.apache.hadoop.fs.*;
+
+public class ParquetStatsExtractor {
+    @Builder.Default
+    private static final ParquetMetadataExtractor parquetMetadataExtractor =
+            ParquetMetadataExtractor.getInstance();
+    @Builder.Default
+    private static final ParquetPartitionExtractor partitionExtractor = ParquetPartitionExtractor.getInstance();
+
+    private static Map<ColumnDescriptor, ColStats> stats = new LinkedHashMap<ColumnDescriptor, ColStats>();
+    private static long recordCount = 0;
+    private Map<String, List<String>> initPartitionInfo() {
+        return getPartitionFromDirectoryStructure(hadoopConf, basePath, Collections.emptyMap());
+    }
+    private InternalDataFile toInternalDataFile(Configuration hadoopConf,
+            String parentPath, Map<ColumnDescriptor, ColStats> stats) {
+        FileSystem fs = FileSystem.get(hadoopConf);
+        FileStatus file = fs.getFileStatus(new Path(parentPath));
+        Map<String, List<String>> partitionInfo = initPartitionInfo();
+
+        ParquetMetadata footer =
+                parquetMetadataExtractor.readParquetMetadata(hadoopConf, parentPath);
+        MessageType schema = parquetMetadataExtractor.getSchema(footer);
+        InternalSchema schema = schemaExtractor.toInternalSchema(schema);
+        List<PartitionValue> partitionValues =
+                partitionExtractor
+                        .getPartitionValue(parentPath,file.getPath().toString(),schema,partitionInfo);
```

Comment:
The partitionValues are not the actual partition values but it's a list of partitioning fields in the schema with their range. 
https://github.com/apache/incubator-xtable/blob/main/xtable-api/src/main/java/org/apache/xtable/model/storage/InternalDataFile.java#L44